temporal-sdk/src/main/java/io/temporal/client/LazyUpdateHandleImpl.java

```diff
@@ -72,12 +73,15 @@ public String getId() {
 
   @Override
   public CompletableFuture<T> getResultAsync(long timeout, TimeUnit unit) {
-    WorkflowClientCallsInterceptor.PollWorkflowUpdateOutput output =
-        workflowClientInvoker.pollWorkflowUpdate(
-            new WorkflowClientCallsInterceptor.PollWorkflowUpdateInput<>(
-                execution, updateName, id, resultClass, resultType, timeout, unit));
+    WorkflowClientCallsInterceptor.PollWorkflowUpdateOutput<T> pollCall;
```

Comment:
Sorry, i have read this a few times and I am not sure logic trying to accomplish?

Reply:
Since before the handle is returned from start when user says complete, there might be a result from polling already and if there is we want to use that, otherwise try it - but then we need to wipe that result in case getResult gets called again

Reply:
hm this seems racy, if you have two concurrent calls isn't is possible for `pollCall=null ` if two threads interleave in the right way?

Reply:
Ah yes it is, I didn't think about this being called concurrently, too used to Rust.

Reply:
I've just sync'd the whole thing since arguably it doesn't make sense to make multiple concurrent calls to server in that case anyway

Reply:
Simplified this